### PR TITLE
Added authentication support

### PR DESCRIPTION
**Pull request by evgenyra**
Hope it will be useful for simple JMX authentication, correct me if I miss something.

### CODE REVIEW

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 1]
```diff
@@ -59,7 +64,9 @@ public void doScrape() throws Exception {
           beanConn = ManagementFactory.getPlatformMBeanServer();
         } else {
           String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
-          jmxc = JMXConnectorFactory.connect(new JMXServiceURL(url), null);
+          HashMap credential = new HashMap();
+          credential.put(username, password );
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: Extra space before )

Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: Is specifying these as empty the same as not specifying them?

Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: yep

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import javax.management.remote.JMXConnector;
 import javax.management.remote.JMXConnectorFactory;
 import javax.management.remote.JMXServiceURL;
+import java.util.HashMap;
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: This should sort earlier in this list

Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: done.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 3]
```diff
@@ -58,9 +64,11 @@ public void doScrape() throws Exception {
         if (hostPort.isEmpty()) {
           beanConn = ManagementFactory.getPlatformMBeanServer();
         } else {
-          String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
-          jmxc = JMXConnectorFactory.connect(new JMXServiceURL(url), null);
-          beanConn = jmxc.getMBeanServerConnection();
+            String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
+            HashMap credential = new HashMap();
+            credential.put(username, password );
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: The extra space is still here. The indentation has also changed.

Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: hm. I am afraid it is my eclipse configuration a little bit different. Could you send me your lang. style settings, then I'll alignment it to you, or in another hand, you can move out my mistakes by the merge.

Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: I use vim, this is formatted by hand.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 4]
```diff
@@ -1,10 +1,13 @@
 package io.prometheus.jmx;
 
 import java.io.IOException;
+
 import java.lang.management.ManagementFactory;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.ArrayList;
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: Keep it sorted

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 5]
```diff
@@ -59,7 +67,16 @@ public void doScrape() throws Exception {
           beanConn = ManagementFactory.getPlatformMBeanServer();
         } else {
           String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
-          jmxc = JMXConnectorFactory.connect(new JMXServiceURL(url), null);
+          HashMap credential = null;
+          if(username != null && username.length() != 0 && password != null && password.length() != 0) {
+        	  credential = new HashMap();
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: Indentation is off here, it should be two spaces

File: collector/src/main/java/io/prometheus/jmx/JmxCollector.java [hunk 6]
```diff
@@ -64,6 +67,20 @@ private JmxCollector(Map<String, Object> config) throws MalformedObjectNameExcep
           hostPort = "";
         }
 
+        if (config.containsKey("username")) {
+            username = (String)config.get("username");
+          } else {
+            // any username
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxCollector.java]: Comments should start with a capital letter and end with a period.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 7]
```diff
@@ -38,11 +42,15 @@ void recordBean(
 
     private MBeanReceiver receiver;
     private String hostPort;
+    private String username;
+    private String password;
     private List<ObjectName> whitelistObjectNames, blacklistObjectNames;
 
-    public JmxScraper(String hostPort, List<ObjectName> whitelistObjectNames, List<ObjectName> blacklistObjectNames,  MBeanReceiver receiver) {
+    public JmxScraper(String hostPort, String username, String password,  List<ObjectName> whitelistObjectNames, List<ObjectName> blacklistObjectNames,  MBeanReceiver receiver ) {
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: There's an extra space after the ,

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 8]
```diff
@@ -271,9 +288,9 @@ public static void main(String[] args) throws Exception {
       List<ObjectName> objectNames = new LinkedList<ObjectName>();
       objectNames.add(null);
       if (args.length > 0) {
-        new JmxScraper(args[0], objectNames, new LinkedList<ObjectName>(), new StdoutWriter()).doScrape();
+        new JmxScraper(args[0], args[1], args[2], objectNames, new LinkedList<ObjectName>(), new StdoutWriter()).doScrape();
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: This will fail with an out of bounds if there's only one argument

**Comment by brian-brazil** [comment]
This generally looks okay. Please address my comments and squash your commits.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java (hunk 4 above)
Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: fixed

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java (hunk 5 above)
Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: fixed

File: collector/src/main/java/io/prometheus/jmx/JmxCollector.java (hunk 6 above)
Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxCollector.java]: fixed

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java (hunk 7 above)
Comment (evgenyra) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: fixed

**Comment by brian-brazil** [comment]
Have you forgotten to push your changes?

**Comment by evgenyra** [comment]
Just placed comments before submissions. :)

**Comment by brian-brazil** [comment]
Two minor comments, then if you can squash your commits I can merge.

**Comment by evgenyra** [comment]
could you describe the line number , if I miss spaces again.

**Comment by brian-brazil** [comment]
Lines 73-76, outdent by 3 spaces.

**Comment by evgenyra** [comment]
fixed.

**Comment by brian-brazil** [comment]
That looks good, can you squash your commits please?

**Comment by evgenyra** [comment]
All commits messages squashed done.

**Comment by brian-brazil** [comment]
Thanks!

**Comment by evgenyra** [comment]
Thank you :)

**Pull request merged**